tools/common/Optimizations.cs

```diff
@@ -188,7 +188,7 @@ public void Initialize (Application app, out List<ProductException> messages)
 				// This means it won't be listed in --help, and it won't be enabled if all optimizations
 				// are enabled. Yet we still might want to enable it manually, and this condition
 				// allows us to manually pass --optimize=remove-dynamic-registrar and enable it that way.
-				if (app.Platform == ApplePlatform.MacOSX && (Opt) i == Opt.RemoveDynamicRegistrar)
+				if (app.Platform == ApplePlatform.MacOSX && app.XamarinRuntime != XamarinRuntime.NativeAOT && (Opt) i == Opt.RemoveDynamicRegistrar)
```

Comment:
I am not sure I understand purpose of this change, so some explanation would be nice. Few lines above it's stated that "dynamic registrar is not supported when using NativeAOT".

Reply:
Note that the if condition is if _not_ using NativeAOT.

This block of code would skip the "remove-dynamic-registrar" optimization on macOS (this is to support "-optimize:all" without showing a warning all the time out not being able to remove the dynamic registrar on macOS).

Skipping the if condition means that we'll apply the "remove-dynamic-registrar" optimization for NativeAOT.

Reply:
Makes sense. The commit message confused me.